sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -503,39 +504,59 @@ export class StudentService extends RecordDataModelService<Student> {
    * Use the information available in the authentication token to update
    * the user and student data currently on DB.
    * @param studentToken student authentication token.
+   * @param options method options:
+   * - `userId` user id provided in case the student data
+   * is updated by the ministry.
    * @returns updated student, if some data was changed.
    */
-  async synchronizeFromUserToken(
-    studentToken: StudentUserToken,
+  async updateStudentUserData(
+    studentUserData: StudentUserData,
+    options?: { userId?: number },
   ): Promise<Student> {
-    const studentToSync = await this.getStudentById(studentToken.studentId);
+    const userId = studentUserData.userId
+      ? studentUserData.userId
+      : options?.userId;
+    const studentToSync = await this.getStudentById(studentUserData.studentId);
     let mustSave = false;
-    if (studentToken.givenNames === undefined) {
-      studentToken.givenNames = null;
+    if (studentUserData.givenNames === undefined) {
+      studentUserData.givenNames = null;
     }
     if (
-      !dayjs(studentToken.birthdate).isSame(studentToSync.birthDate) ||
-      studentToken.lastName !== studentToSync.user.lastName ||
-      studentToken.givenNames !== studentToSync.user.firstName
+      !dayjs(studentUserData.birthdate).isSame(studentToSync.birthDate) ||
+      studentUserData.lastName.toLowerCase() !==
+        studentToSync.user.lastName.toLowerCase() ||
+      studentUserData.givenNames.toLowerCase() !==
```

Comment:
I believe the `null` check is missing below. Should it be changed as below?
```ts
studentUserData.givenNames?.toLowerCase() !== studentToSync.user.firstName?.toLowerCase()
```

Reply:
Not needed from my understanding. `"".toLowerCase()` will be `""`